src/Controllers/CalculatorController.cpp

```diff
@@ -0,0 +1,125 @@
+#include "CalculatorController.h"
+
+#include <string>
+#include <unordered_map>
+#include <iostream>
+
+#include <pistache/router.h>
+#include <pistache/http.h>
+#include <rapidjson/document.h>
+
+#include "CalculatorService.h"
+#include "Result.h"
+#include "HistoryService.h"
+#include "CalcHistoryPair.h"
+
+
+
+static void sendError(Pistache::Http::ResponseWriter& res, std::string message)
+{
+    res.send(Pistache::Http::Code::Bad_Request,
+                 message.c_str(),
+                 Pistache::Http::Mime::MediaType(
+                     Pistache::Http::Mime::Type::Text,
+                     Pistache::Http::Mime::Subtype::Plain));
+    return;
+};
+
+
+
+void CalculatorController::calculate(const Pistache::Rest::Request& req, Pistache::Http::ResponseWriter res)
```

Comment:
Instead of manually validating the body json, I think there's a way that you can have the flatbuffers library do it for you, or, at least, part of it for you. In Dune, we usually just work with the flatbuffers directly, and the CDM handles conversion from the flatbuffer types to the json. I found something on the following flatbuffers page that might be helpful here: https://google.github.io/flatbuffers/flatbuffers_guide_use_cpp.html (see the "Making your program capable of loading text directly" sub-section). There's also an example over at https://github.com/google/flatbuffers/blob/master/samples/sample_text.cpp that loads json into flatbuffer objects dynamically at runtime. That way you can might be able to offload some of the parsing onto the flatbuffers library.